src/queue/consumers/oneidentity/utils/OneIdentityApi.ts

```diff
@@ -0,0 +1,107 @@
+import axios, { AxiosInstance, AxiosResponse } from 'axios';
+
+interface CreateEntityResult {
+  uid?: string;
+  uri?: string;
+}
+
+interface EntityValues<T> {
+  values: T;
+}
+
+// This class is a basic wrapper around the OneIdentity API
+export class OneIdentityApi {
+  private baseUrl: string;
+  private apiUrl: string;
+  private axiosInstance: AxiosInstance;
+
+  constructor(url: string) {
+    this.baseUrl = url;
+    this.apiUrl = `${this.baseUrl}/api`;
+    this.axiosInstance = this.createAxiosInstance();
+  }
+
+  private createAxiosInstance(): AxiosInstance {
```

Comment:
I am wondering if we should use native node fetch here, as we are using node 18+ anyway, instead of axios. Maybe I am wrong but I was thinking it would be nice if we can completely remove using external libraries for this in the future as node new versions supports it out of the box.

Reply:
Yeah, I saw Axios being used somewhere else, so I used it too. We can replace it everywhere in the future.

Reply:
Yes we can have a separate ticket for this.